java/code/src/com/redhat/rhn/manager/system/SystemManager.java

```diff
@@ -1175,7 +1175,8 @@ public static PagedSqlQueryBuilder virtualSystemsListQueryBuilder() {
                 .from("rhnVirtualInstance VI " +
                         "    LEFT OUTER JOIN rhnVirtualInstanceInfo VII ON VI.id = VII.instance_id " +
                         "    LEFT OUTER JOIN rhnVirtualInstanceState VIS ON VII.state = VIS.id " +
-                        "    LEFT OUTER JOIN suseSystemOverview S ON S.id = VI.virtual_system_id")
+                        "    LEFT OUTER JOIN suseSystemOverview S ON S.id = VI.virtual_system_id" +
+                        "    LEFT OUTER JOIN rhnServer RS ON RS.id = VI.host_system_id")
```

Comment:
The problem with the current query is that you wouldn't be able to add a where clause to filter the host_server_name like:
`and host_server_name like '%name%'`
but it would work for the server_name:
`and server_name like '%name%'`

Having a subquery in the select doesnt add it as part of the result set columns, while coalesce does it.
Thats why i added the host_server_name as a coalesce column as well